LocalLang/Evaluator.lean

```diff
@@ -1,5 +1,39 @@
 import LocalLang.AST
 
--- Maybe `m ℕ` for some monad `m`?
--- TODO: implement
-def evaluate : Program → ℕ := sorry
+abbrev Vars := Std.HashMap String ℕ
+
+def BinOp.eval : BinOp → ℕ → ℕ → ℕ
+  | add, x, y => x + y
+  | mul, x, y => x * y
+
+def Function.inline (V : Vars) : Function → List ℕ → (Std.HashMap String ℕ × Expr)
```

Comment:
Could you add a comment describing what this does?  It isn't super clear for me.
Also, is it really worth making this one function?  It feels to me like just the V_with_args part is important; the other part is just `fun f => f.body`.

nit: Use `Vars` for result as well, please